### PR TITLE
Add `KeepAliveClientRequest` class for k8s async client

### LINKED ISSUE #14954: K8s worker fails monitoring flow and sets it to crashed.
### Bug summary

We are seeing an issue with the latest k8s worker monitoring flow runs. After some amount of time (5-10 minutes?) we see the worker fail to monitor the flow run with the message provided in additional context. The worker then marks the flow as failed. The flow continues to run until it tries to update its task status at which point it fails because its task status is already in a terminal state.

### Version info (`prefect version` output)

```Text
prefect-client: 2.20.0
prefect-kubernetes: 0.4.1
chart version: 2024.8.3144127
```

### Additional context

```
Error during task execution: Response payload is not completed: <TransferEncodingError: 400, message='Not enough data for satisfy transfer length header.'>
NoneType: None
```

### PR DESCRIPTION
**Pull request by kevingrismore**
Closes #14954 

Swaps the request class on the async k8s base client for one that has TCP keepalive enabled after the client is built. Refer to https://github.com/aio-libs/aiohttp/issues/3904#issuecomment-759205696

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/integrations/prefect-kubernetes/prefect_kubernetes/utilities.py [hunk 1]
```diff
@@ -14,34 +16,35 @@
...
-    messages. Thus the flags are added to be used on the underlying sockets.
+    aiohttp only directly implements socket keepalive for incoming connections
+    in its RequestHandler. For client connections, we need to set the keepalive
+    ourselves.
+
+    Refer to https://github.com/aio-libs/aiohttp/issues/3904#issuecomment-759205696
     """
 
-    socket_options = [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)]
+    async def send(self, conn: "Connection") -> "ClientResponse":
```
Comment (zzstoatzz) [inline src/integrations/prefect-kubernetes/prefect_kubernetes/utilities.py:28]: if we're actually importing these, we should be able to use them
```suggestion
    async def send(self, conn: Connection) -> ClientResponse:
```

Comment (kevingrismore) [reply src/integrations/prefect-kubernetes/prefect_kubernetes/utilities.py:28]: true

**Comment by kevingrismore** [comment]
To be clear about why I opened this, `client.rest_client.pool_manager.connection_pool_kw` in the previous implementation is accessing a property of a `urllib3` connection pool. `aiohttp` has a different object structure and we need to handle the keepalive configuration slightly differently.

**Review by zzstoatzz — APPROVED** [review]
🧼

**Comment by kevingrismore** [comment]
Before: crash with error matching reports when run in a cluster with a 4 minute TCP load balancing timeout rule
image
```
Error occurred while streaming logs - Job will continue to run but logs will no longer be streamed to stdout.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client_proto.py", line 94, in connection_lost
    uncompleted = self._parser.feed_eof()
                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "aiohttp/_http_parser.pyx", line 513, in aiohttp._http_parser.HttpParser.feed_eof
aiohttp.http_exceptions.TransferEncodingError: 400, message:
  Not enough data for satisfy transfer length header.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/prefect_kubernetes/worker.py", line 939, in _stream_job_logs
    async for line in logs.content:
  File "/usr/local/lib/python3.11/site-packages/aiohttp/streams.py", line 50, in __anext__
    rv = await self.read_func()
         ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/streams.py", line 317, in readline
    return await self.readuntil()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/streams.py", line 351, in readuntil
    await self._wait("readuntil")
  File "/usr/local/lib/python3.11/site-packages/aiohttp/streams.py", line 312, in _wait
    await waiter
aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed: <TransferEncodingError: 400, message='Not enough data for satisfy transfer length header.'>. ConnectionResetError(104, 'Connection reset by peer')
```

After: no crash, flow completes successfully
image

**Pull request merged**